[EASI-4705] Set up GRB review form wrapper

aterstriep opened the pull request:
<!--
REQUIRED
    Ensure that your PR title has the relevant Jira ticket number(s) in the title.
    Follow this pattern: [EASI-1234] [EASI-4567] Title of the PR.
    Use [NOREF] in place of a ticket number when there's no associated Jira ticket.
    The heading below should just have the ticket numbers (for easy linking in Jira)
-->
# EASI-4705

## Description
- Created form wrapper for GRB review form
  - Stepped header disables steps until previous required fields are completed
    - Redirects user if trying to access a disabled step
  - If fields are valid, form will submit when using "save and return" buttons or navigating to new step
- Created components for each step
  - Review type step is partially done to make it easier to test basic form wrapper functionality, but is not part of this ticket
- Unit tests


## How to test this change
- Navigate to form (`/it-governance/:systemId/grb-review/review-type`)
- Test form wrapper and stepped header functionality
  - To enable all steps, comment out [this line](https://github.com/CMS-Enterprise/easi-app/blob/d006bdcba257a2abaef75022d0dac285b87d555c/src/features/ITGovernance/Admin/GRBReview/GRBReviewForm/GRBReviewFormStepWrapper/index.tsx#L140)

## PR Author Checklist
<!--
REQUIRED
    Ensure that each of the following is true before you submit this PR (or before it leaves "draft" status), and check each box to confirm
-->

- [x] I have provided a detailed description of the changes in this PR.
- [x] I have provided clear instructions on how to test the changes in this PR.
- [x] I have updated tests or written new tests as appropriate in this PR.


## PR Reviewer Guidelines
<!--
This is just some static content to ensure we're following best practices when reviewing.
There is no need to edit this section.
-->
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

garyjzhao commented on src/features/ITGovernance/Admin/GRBReview/GRBReviewForm/ReviewType/index.tsx:
```diff
@@ -0,0 +1,88 @@
+      >
+        <FormGroup error={!!errors.grbReviewType} className="margin-top-5">
+          <Fieldset>
+            <legend className="text-bold">
+              {t('setUpGrbReviewForm.reviewType.label')} <RequiredAsterisk />
+            </legend>
```
Hi 👋🏼  was just looking around the code while it was still in draft mode. 
Truss has a built in "required marker" to their `Label` component. 
To use: `<Label requiredMarker />`

Label props below:
```
declare type LabelProps = {
    children: React.ReactNode;
    htmlFor: string;
    className?: string;
    error?: boolean;
    hint?: React.ReactNode;
    srOnly?: boolean;
    requiredMarker?: boolean;
};
```

aterstriep replied:
Thanks for pointing that out! Looks like that prop wasn't available before we upgraded `react-uswds`, so I hadn't noticed the change. This is a `legend` instead of `label`, but I see that the `Fieldset` component also added a `requiredMarker` prop that gets added to the `legend`.

I usually just use a separate `legend` element because there isn't a way to make the `Fieldset` component's legend bold without either adding CSS or wrapping it in a `span` tag within the `legend` prop.

patrickseguraoddball commented on src/gql/operations/ITGov/GRBReview/UpdateSystemIntakeGRBReviewType.ts:
```diff
@@ -0,0 +1,15 @@
+import { gql } from '@apollo/client';
+import SystemIntakeGRBReview from 'gql/operations/Fragments/SystemIntakeGRBReview';
+
+export default gql(/* GraphQL */ `
+  ${SystemIntakeGRBReview}
+  mutation UpdateSystemIntakeGRBReviewType(
+    $input: updateSystemIntakeGRBReviewTypeInput!
+  ) {
+    updateSystemIntakeGRBReviewType(input: $input) {
+      systemIntake {
+        ...SystemIntakeGRBReview
+      }
+    }
+  }
+`);
```
Same as above

patrickseguraoddball reviewed (approved):
Nice work Ashley :). Works great

Only thing I noticed is that for `Save and exit` we do a `isDirty` check before needing to call the mutation, however we don't do that for normal form progression.  This isn't a huge issue at all, fine with approving as is.  The codegen updates are really blocking either.

aterstriep commented on src/gql/operations/ITGov/GRBReview/GetSystemIntakeGRBReview.ts:
```diff
@@ -0,0 +1,11 @@
+      ...SystemIntakeGRBReview
+    }
+  }
+`);
```
I didn't know that - thanks for pointing it out!

aterstriep commented:
> Only thing I noticed is that for Save and exit we do a isDirty check before needing to call the mutation, however we don't do that for normal form progression.

@patrickseguraoddball I fixed this and also added an `isValid` check on the Autosave so that it doesn't keep running failed mutations. I also updated the queries to remove the fragment imports.

patrickseguraoddball reviewed (approved):
LG Thanks!
